### PR TITLE
Add support for default user modes. #942

### PR DESCRIPTION
**Pull request by ajaspers**

### CODE REVIEW

File: irc/client.go [hunk 1]
```diff
@@ -318,6 +318,10 @@ func (server *Server) RunClient(conn clientConn, proxyLine string) {
 	session.idletimer.Initialize(session)
 	session.resetFakelag()
 
+	for _, defaultMode := range config.Accounts.defaultUserModes {
```
Comment (slingamn) [inline irc/client.go:321]: Until we get support for persisting user modes of always-on clients in #819, can you copy and paste this block into `AddAlwaysOnClient` (the other code path that creates a new client)?

File: oragono.yaml [hunk 2]
```diff
@@ -451,6 +451,12 @@ accounts:
         offer-list:
             #- "oragono.test"
 
+    # modes that are set by default when a user connects
```
Comment (slingamn) [inline oragono.yaml:454]: This is extra lame, but can you copy and paste this block into conventional.yaml (in the same place it appears in oragono.yaml)?

**Comment by slingamn** [comment]
@DanielOaks any thoughts on making `+i` a recommended default (i.e., filling it in in the default `oragono.yaml` file, but continuing to leave it unset for older config files that omit the field)?

File: irc/client.go (hunk 1 above)
Comment (ajaspers) [reply irc/client.go:321]: Done.

File: oragono.yaml (hunk 2 above)
Comment (ajaspers) [reply oragono.yaml:454]: Done.

**Comment by DanielOaks** [comment]
@slingamn definitely support `+i` being a recommended default. Thanks for the work on this, looks great!

**Pull request merged**